toolchains/solidity/core/crates/slither-server/src/slither.rs

```diff
@@ -1,50 +1,54 @@
+use crate::{error::SlitherError, utils::normalize_slither_path, types::SlitherResult};
+use std::process::Stdio;
+use tokio::{io::AsyncReadExt, process::Command};
 use tower_lsp::lsp_types::Diagnostic;
 
-pub fn is_slither_installed() -> bool {
-    let output = Command::new("slither").arg("--version").output();
-    output.is_ok()
-}
 
-pub fn is_solc_installed() -> bool {
-    let output = Command::new("solc").arg("--version").output();
-    output.is_ok()
-}
+pub async fn parse_slither_out(uri: &str) -> Result<Vec<Diagnostic>, SlitherError> {
+    let mut results: Vec<Diagnostic> = Vec::new();
 
-#[cfg(target_family = "windows")]
-fn normalize_slither_path(path: &str) -> String {
-    let mut path = path.replace("%3A/", "://");
-    path.remove(0);
-    path.to_string()
-}
+    eprintln!("SLITHER STARTING");
+    let mut output = exec_slither(uri)?;
+    let out = match output.stdout.take() {
+        Some(out) => out,
+        None => return Err(SlitherError::Unknown("Failed to get slither output pipe".to_string())),
+    };
 
-#[cfg(not(target_family = "windows"))]
-fn normalize_slither_path(path: &str) -> String {
-    path.to_string()
-}
+    let mut buffer = tokio::io::BufReader::new(out);
+    let mut dst = String::new();
 
-pub fn exec_slither(filepath: &str) -> Result<Vec<Diagnostic>, SlitherError> {
-    let mut results: Vec<Diagnostic> = Vec::new();
-    let out = Command::new("slither")
-        .arg(normalize_slither_path(filepath))
-        .arg("--exclude")
-        .arg("naming-convention")
-        .arg("--json")
-        .arg("-")
-        .output()?;
-    if out.status.code() == Some(1) {
-        eprintln!("Unknown error occured: {:?}", out);
-        return Err(SlitherError::Unknown);
-    }
-    if out.stdout.is_empty() {
-        return Ok(results);
-    }
-    let json: SlitherResult =
-        serde_json::from_str(&String::from_utf8_lossy(&out.stdout).replace("\\\"", "\""))?;
-    for detector in json.results.detectors {
-        results.append(&mut crate::types::diag_from_json(detector.clone()));
+    output.wait().await?;
+    eprintln!("SLITHER FINISHED");
+    
+    buffer.read_to_string(&mut dst).await?;
+    let json: Result<SlitherResult, serde_json::Error> =
+    serde_json::from_str(&dst);
+    
+    match json {
+        Ok(json) => {
+            for detector in json.results.detectors {
+                results.append(&mut crate::types::diag_from_json(detector.clone()));
+            }
+        }
+        Err(e) => {
+            eprintln!("Error parsing slither output: {}", e);
+            return Err(SlitherError::ParsingFailed(e));
+        }
     }
+
     Ok(results)
 }
+
+
+fn exec_slither(uri: &str) -> Result<tokio::process::Child, std::io::Error> {
+    Command::new("slither")
+    .arg(normalize_slither_path(uri))
+    .arg("--exclude")
+    .arg("naming-convention")
+    .arg("--json")
+    .arg("-")
+    .stdout(Stdio::piped())
+    .stderr(Stdio::piped())
+    .stdin(Stdio::null())
+    .spawn()
```

Comment:
should read the .slither.json to know which rules to exclude, remappings and other things

Reply:
You should check if slither loads by default the slither config and if it does specify the directory to run slither on

Reply:
For the moment slither runs on targeted files, again nice feature to have yes, but lets move on and make another PR for that, wdyt @0xMemoryGrinder ?

Reply:
Check if slither uses the config file by default (without additional argument). If it does, just add a cwd in the command with the workspace root as the value. 

If not it can be evicted from this PR scope, the main goal is to close this PR ASAP to finally make a proper release